script.js

```diff
@@ -1,40 +1,84 @@
-function colorClick(event) {
+function colorSelected({ target }) {
   const selectedColor = document.querySelector('.selected');
-  selectedColor.classList.remove('selected');
-  event.target.classList.add('selected');
-
-  /*
-    Não me recordava como adicionar uma classe e ao buscar no google encontrei:
-    https://pt.stackoverflow.com/questions/225809/como-adicionar-uma-classe-em-javascript-puro */
+  selectedColor.classList.toggle('selected');
+  target.classList.add('selected');
 }
 
-function pixelClick(event) {
-  const selectedColor = document.querySelector('.selected');
-  event.target.style.backgroundColor =
-    getComputedStyle(selectedColor).backgroundColor;
+function colorPixel(event) {
+  const selectedBackground = getComputedStyle(document.querySelector('.selected')).backgroundColor;
+  const { target } = event;
+  const currentPixelColor = getComputedStyle(target).backgroundColor;
 
-  /* 
-    https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle 
-  */
+  if (currentPixelColor === selectedBackground) {
+    target.style.backgroundColor = 'white';
+  } else {
+    target.style.backgroundColor = selectedBackground;
+  }
 }
 
-function clearColors(event) {
+function clearBoard() {
   const pixels = document.querySelectorAll('.pixel');
   for (let index = 0; index < pixels.length; index += 1) {
     pixels[index].style.backgroundColor = 'white';
   }
 }
 
-const colors = document.getElementsByClassName('color');
+function validateBoard(boardSize) {
+  if (boardSize < 5) return 5;
+  if (boardSize > 50) return 50;
+  if (Number.isNaN(boardSize)) return false;
```

Comment:
Da pra chamar direto aqui se quiser

```suggestion
  if (isNaN(boardSize)) return false;
```

Reply:
Eu fiz desse jeito por causa do lint. Ele pedia para utilizar o Number.